Add dg kernels to MFEM backend (#32267)
idaholab/moose: framework/doc/content/source/mfem/bcs/MFEMDGDiffusionBC.md

# MFEMDGDiffusionBC

!if! function=hasCapability('mfem')

## Overview

> Adds a boundary face integrator to the bilinear form. The DG parameters sigma and kappa can be set, but both have default values of

Reviewer: Needs updating, e.g. we now add both a bilinear and a linear integrator.
If we don't want to type the integrators' forms, we should at least link to somewhere the user can find them in the mfem docs. Same for the kernel documentation.

# MFEMDGDiffusionBC

!if! function=hasCapability('mfem')

## Overview

Adds a boundary face integrator to the linear/bilinear forms.

> `createBFIntegrator` returns a `mfem::DGDiffusionIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDiffusionIntegrator.html)).

Reviewer: ```suggestion
`createBFIntegrator()` returns an `mfem::DGDiffusionIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDiffusionIntegrator.html)).
```

> `createLFIntegrator` returns a `mfem::DGDirichletLFIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDirichletLFIntegrator.html)).

Reviewer: ```suggestion
`createLFIntegrator()` returns an `mfem::DGDirichletLFIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDirichletLFIntegrator.html)).
```

The DG parameters sigma and kappa can be set, but both have default values of

> -1.0 and (order+1)^2 respectively.  A value of $\sigma = -1$ corresponds to a symmetric interior penalty DG method.

Reviewer: ```suggestion
The DG parameters $\sigma$ and $\kappa$ can be set, but both have default values of
$\sigma = -1$ and $\kappa = (o+1)^2$, respectively. $o$ is the order of the finite element space of the variable this BC applies to.
A value of $\sigma = -1$ corresponds to a symmetric interior penalty DG method.
```

## Example Input File Syntax

!listing test/tests/mfem/kernels/dg_diffusion.i block=BCs

!syntax parameters /BCs/MFEMDGDiffusionBC

!syntax inputs /BCs/MFEMDGDiffusionBC

> 

Reviewer: ```suggestion

!syntax children /BCs/MFEMDGDiffusionBC
```

Reviewer: When we tried addressing this, we actually deleted the inputs.

# MFEMDGDiffusionBC

!if! function=hasCapability('mfem')

## Overview

Adds a boundary face integrator to the linear/bilinear forms.
`createBFIntegrator()` returns an `mfem::DGDiffusionIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDiffusionIntegrator.html)).

> `createLFIntegrator()` returns an `mfem::DGDirichletLFIntegrator` object ([see mfem docs for more info](https://docs.mfem.org/html/classmfem_1_1DGDirichletLFIntegrator.html)).

Reviewer: ```suggestion
`createBFIntegrator()` returns an [`mfem::DGDiffusionIntegrator`](https://docs.mfem.org/html/classmfem_1_1DGDiffusionIntegrator.html).
`createLFIntegrator()` returns an [`mfem::DGDirichletLFIntegrator`](https://docs.mfem.org/html/classmfem_1_1DGDirichletLFIntegrator.html).
```
